identity/app/form/PrivacyMapping.scala

```diff
@@ -1,51 +1,81 @@
+        mapping(
+          "actor" -> text,
+          "consentIdentifier" -> text,
+          "consentIdentifierVersion" -> number,
+          "hasConsented" -> boolean,
+          "timestamp" -> jodaDate(dateTimeFormatISO8601),
+          "privacyPolicy" -> number
+        )(Consent.apply)(Consent.unapply)
+      )
+    )(PrivacyFormData.apply)(PrivacyFormData.unapply)
 
-  protected lazy val contextMap =  Map(
+  protected def toUserFormData(userFromApi: User): PrivacyFormData =
+    PrivacyFormData(userFromApi)
+
+  protected lazy val idapiErrorContextToFormFieldKeyMap =  Map(
     "statusFields.receiveGnmMarketing" -> "receiveGnmMarketing",
     "statusFields.receive3rdPartyMarketing" -> "receive3rdPartyMarketing",
     "statusFields.allowThirdPartyProfiling" -> "allowThirdPartyProfiling"
   )
 }
 
+/**
+  * Form specific DTO representing marketing consent subset of User model
+  */
 case class PrivacyFormData(
-                            receiveGnmMarketing: Boolean,
-                            receive3rdPartyMarketing: Boolean,
-                            allowThirdPartyProfiling: Boolean
-                            ) extends UserFormData{
-
-  def toUserUpdate(currentUser: User): UserUpdate = {
-    val statusFields = currentUser.statusFields
-    UserUpdate(
-      statusFields = Some(statusFields.copy(
+    receiveGnmMarketing: Boolean,
+    receive3rdPartyMarketing: Boolean,
+    allowThirdPartyProfiling: Boolean,
+    consents: List[Consent]) extends UserFormData{
+
+  def toUserUpdate(oldUserFromApi: User): UserUpdateDTO =
+    UserUpdateDTO(
+      statusFields = Some(oldUserFromApi.statusFields.copy(
         receive3rdPartyMarketing = Some(receive3rdPartyMarketing),
         receiveGnmMarketing = Some(receiveGnmMarketing),
         allowThirdPartyProfiling = Some(allowThirdPartyProfiling)
-      ))
+      )),
+      consents = Some(consents)
     )
-  }
-
 }
 
 object PrivacyFormData {
-  def apply(user: User): PrivacyFormData = PrivacyFormData(
-    receiveGnmMarketing = user.statusFields.receiveGnmMarketing.getOrElse(false),
-    receive3rdPartyMarketing = user.statusFields.receive3rdPartyMarketing.getOrElse(false),
-    allowThirdPartyProfiling = user.statusFields.allowThirdPartyProfiling.getOrElse(true)
-  )
+  /**
+    * Converts User DO from IDAPI to form processing DTO PrivacyFromData
+    *
+    * @param userFromApi Identity User domain model from IDAPI defiend in identity-model library
+    * @return form processing DTO PrivacyFromData
+    */
+  def apply(userFromApi: User): PrivacyFormData =
+    PrivacyFormData(
+      receiveGnmMarketing = userFromApi.statusFields.receiveGnmMarketing.getOrElse(false),
+      receive3rdPartyMarketing = userFromApi.statusFields.receive3rdPartyMarketing.getOrElse(false),
+      allowThirdPartyProfiling = userFromApi.statusFields.allowThirdPartyProfiling.getOrElse(true),
+      consents = if (userFromApi.consents.isEmpty) defaultConsents else userFromApi.consents
```

Comment:
I kind of thought that `Consents` would be an `Option`, but I suppose that depends on how it is handled by the model.